client/payments/data/action-types.js

```diff
@@ -0,0 +1,7 @@
+const TYPES = {
+	RECEIVE_PAYMENT_GATEWAY: 'RECEIVE_PAYMENT_GATEWAY',
```

Comment:
I'd love for our data stores to be more event-driven going forward.  I know we have a lot of these patterns in the existing data store, but Redux recommends that actions are more "event-like" than describing them as actions.

This action would be broken into:
`GET_PAYMENT_GATEWAYS_REQUEST`
`GET_PAYMENT_GATEWAYS_SUCESS`
`GET_PAYMENT_GATEWAYS_ERROR`

Some prior art in https://github.com/woocommerce/woocommerce-admin/pull/6275/files

Reply:
@joshuatf I agree they should be event-like, but I think this is event-like, its just not as granular as what you've suggested. I think the idea of separating out events for success and error statuses is good, it will be easier to read and write a reducer that handles these as separate events, and resolvers will be easier to reason about too :+1:

One question though: What is the benefit of the `REQUEST` event? In a resolver we already have a concept of determining if a request is in flight? What state do we need to track in that case that the built in helpers don't already track?

Reply:
Thank you @joshuatf I will work on separating them 👍

Reply:
@joshuatf I've made some [changes](https://github.com/woocommerce/woocommerce-admin/pull/6918/files). I found myself repeating with those three actions for each action & resolver (at least for this particular use case). I added `SET_ERROR` and `SET_IS_REQUESTING` instead. What do think?

Reply:
I would try to avoid action types prefixed with verbs as these are not very event-like.

Maybe instead these become `UPDATE_PAYMENT_GATEWAY_ERROR` and `UPDATE_PAYMENT_GATEWAY_REQUEST`.  The latter (the request event) kicks off other changes around the request, such as setting `isRequesting` to `true` in the reducer.